### ISSUE TITLE
read variables back into the program

### ISSUE THREAD

**Issue by ClundXIII** [dstahlke/gnuplot-iostream#14]
Hi,

I need to do a few (~ 650k) fits from within my program. Is it possible to automatically read the result back into a variable? Or should I rather use the higher-level interface for this case?

Regards,
Simon

**Comment by dstahlke** [comment]
No, this is not possible currently.  My interface doesn't currently have 
the ability to receive feedback back from gnuplot.  It is able to 
capture mouse clicks, but that works by telling gnuplot to send that 
info to a file (technically, a pty) and I read it from there.

It would be feasible to add this feature, but it'd take some work.  Even 
if such a thing did work though, I think it'd probably be better to just 
do the fit from within your program (not through gnuplot) using some 
library designed for that purpose.  Any linear algebra package would 
make it easy to compute a linear fit, or even fit a linear combination 
of nonlinear functions, e.g., a polynomial.  For more general fitting, I 
know such a library exists but I can't remember the name of it.

- Dan

**Comment by ClundXIII** [comment]
Okay, thanks.
I will most likely use dlib for fitting. The curve is unfortunately not trivial.

**Comment by MohamedLamineAllal** [comment]
Armadillo is a nice library to check.
http://arma.sourceforge.net/
